app/facility/[id]/profile/components/facility-profile-form.tsx

```diff
@@ -0,0 +1,234 @@
+import { Alert } from "@/app/components/ui/alert";
+import { FacilityProfile } from "@prisma/client";
+import { TypeOf, z } from "zod";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import {
+  Form,
+  FormControl,
+  FormLabel,
+  FormMessage,
+  FormField,
+  FormItem,
+} from "@/app/components/ui/form";
+import { Loader, XIcon } from "lucide-react";
+import axios from "axios";
+import { useToast } from "@/app/components/ui/use-toast";
+import Link from "next/link";
+
+const profileSchema = z.object({
+  name: z.string().min(1, "Name is required"),
+  facilityType: z.string().min(1, "Facility Type is required"),
+  description: z.string().optional(),
+  address: z.string().min(10, "Address is required"),
+});
+type ProfileFormValues = z.infer<typeof profileSchema>;
+
+const FacilityProfileForm = ({
+  profile,
+  userId,
+}: {
+  profile: FacilityProfile | null;
+  userId: string;
+}) => {
+  const defaultValues: Partial<ProfileFormValues> = {
+    name: profile?.name ?? "",
+    facilityType: profile?.facilityType ?? "",
+    description: profile?.description ?? "",
+    address: profile?.address ?? "",
+  };
+  const form = useForm<ProfileFormValues>({
+    resolver: zodResolver(profileSchema),
+    defaultValues,
+  });
+  const [profileImageFile, setProfileImageFile] = useState<null | File>();
+  const [profileImageUrl, setProfileImageUrl] = useState(profile?.profileImage);
+  const { toast } = useToast();
+
+  const handleProfileImageChange = (event: any) => {
+    const file = event.target.files[0];
+    if (file && /^(image\/jpeg|image\/png|image\/svg\+xml)$/.test(file.type)) {
+      setProfileImageFile(file);
+    } else {
+      alert("Only JPEG, PNG, or SVG files are allowed.");
+      event.target.value = "";
+    }
+  };
+  const onSubmit = async (data: ProfileFormValues) => {
+    const formData = new FormData();
+    formData.append("name", data.name);
+    formData.append("address", data.address);
+    formData.append("description", data.description ?? "");
+    formData.append("facilityType", data.facilityType);
+    if (profileImageFile) formData.append("profileImage", profileImageFile);
+
+    try {
+      const res = await axios.post(`/api/facility/${userId}`, formData, {
+        headers: {
+          "Content-Type": "multipart/form-data",
+        },
+      });
+      setProfileImageUrl(res.data.profile.profileImage);
+      setProfileImageFile(null);
+      toast({
+        title: "Profile Updated Successfully",
+        variant: "default",
+      });
+    } catch (error: any) {
+      console.error(error);
+      toast({
+        title: "Server Side Error",
+        description: error?.response?.statusText,
+        variant: "destructive",
+      });
+    }
+  };
+
+  const { errors, isSubmitting } = form.formState;
+  return (
+    <Form {...form}>
+      <form
+        className="w-full max-w-3xl py-4"
+        onSubmit={form.handleSubmit(onSubmit)}
+      >
```

Comment:
Are we not able to use the *Form* component for those props?

```suggestion
    <Form {...form} className="w-full max-w-3xl py-4"
        onSubmit={form.handleSubmit(onSubmit)}>
```